From 8c8c82e65815a40a4560fd74eecf2ca42c3d1a3b Mon Sep 17 00:00:00 2001
From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Tue, 11 Feb 2025 09:20:44 +0100
Subject: Revert "hostfs: Add const qualifier to host_root in hostfs_fill_super()"

From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

This reverts commit 1fbe93dd7e6a0dab59a5fce99f16703f7d52ed81 which is
commit 104eef133fd9c17e4dc28bf43f592a86f26d8a59 upstream.

It is reported to cause build issues and odds are the root problem isn't
really an issue on the 6.6.y branch anyway.  If it is, someone can
provide a working set of backported patches.

Reported-by: Guenter Roeck <linux@roeck-us.net>
Link: https://lore.kernel.org/r/cd10a924-ae65-4b02-aea2-e629947ca7a3@roeck-us.net
Cc: Hongbo Li <lihongbo22@huawei.com>
Cc: Christian Brauner <brauner@kernel.org>
Cc: Sasha Levin <sashal@kernel.org>
Cc: Nathan Chancellor <nathan@kernel.org>
Cc: Richard Weinberger <richard@nod.at>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 fs/hostfs/hostfs_kern.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/fs/hostfs/hostfs_kern.c
+++ b/fs/hostfs/hostfs_kern.c
@@ -925,8 +925,8 @@ static const struct inode_operations hos
 static int hostfs_fill_super(struct super_block *sb, struct fs_context *fc)
 {
 	struct hostfs_fs_info *fsi = sb->s_fs_info;
-	const char *host_root = fc->source;
 	struct inode *root_inode;
+	char *host_root = fc->source;
 	int err;
 
 	sb->s_blocksize = 1024;
